server/src/main/java/org/elasticsearch/action/admin/indices/analyze/TransportAnalyzeAction.java

```diff
@@ -163,7 +164,9 @@ private static Analyzer getAnalyzer(AnalyzeAction.Request request, AnalysisRegis
                 if (analyzer == null) {
                     throw new IllegalArgumentException("failed to find global analyzer [" + request.analyzer() + "]");
                 }
-                return analyzer;
+                return analyzer instanceof NamedAnalyzer
+                    ? new NamedAnalyzer((NamedAnalyzer) analyzer, TextFieldMapper.Defaults.POSITION_INCREMENT_GAP)
```

Comment:
This affects all analyzers, not just `standard`, correct?  A minor issue I see here is that we're effectively creating a new instance every time the API is called, instead of re-using existing instances, especially as the override we want to apply is constant and independent  of the request itself. 

Maybe we could somehow cache the result and reuse it for future calls to avoid constantly creating new objects?

Reply:
Thanks, @pmpailis, for the review. That's true; it affects all the available analyzers. I experimented with the following code using a ConcurrentHashMap at the TransportAnalyzeAction layer.

```
   return analyzer instanceof NamedAnalyzer
                    ? cachedNamedAnalyzer.computeIfAbsent(
                        request.analyzer(),
                        (key) -> new NamedAnalyzer((NamedAnalyzer) analyzer, TextFieldMapper.Defaults.POSITION_INCREMENT_GAP)
                    )
                    : analyzer;
                }
```

Reply:
Any cache would then potentially cause a leak. we would have to have a size limit and it would have to be periodically cleared. It would be crazy to some body to call `_analyze` a bunch but then we create a many separate objects, seems really bad.

Reply:
A new named instance that is then GC'd seems safe?

Reply:
Sorry for drive-by commenting, but looked with @drempapis at a few questions here earlier today and just want to note that I'd also say that I think caching is probably overcomplicating things here. IMHO the "_analyze" API in its vanilla-no-index version shouldn't be called that heavily to really warrant such an additional indirection.

Reply:
++ that's fair! Thanks @benwtrent and @cbuescher ! I remember seeing some pretty huge numbers for `_analyze` endpoint in some SDHs hence the suggestion (although not sure how to distinguish if there was an index attached or not), but you obviously have a much better understanding of potential implications with this, if any at all!  Marking this as resolved.

Reply:
@pmpailis if this was doing something more than wrapping an existing object, I would agree that the cost might warrant more complexity.

But the instance here is nothing more than some object refs.

Reply:
yeap, got that! my suggestion was merely on the fact that we know in advance that we can only generate a handful of different instances so, there would be no need to instantiate new objects (taking into account some large `_analyze` numbers as well). But I can see how that might actually overcomplicate things for no real benefit in this scenario.